From c1475020bd4d8bb306fac33410ac5517cf2008f4 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 30 Jan 2017 14:44:12 +0000
Subject: [PATCH 0748/1000] MANTIS 44425 mantis 44425: ConveyorDev.idl with
 more status details - Interface version 1.13 - Define struct baggageStatus
 with details for insertionBelt, verificationBelt, parkingBeltAirline, and
 parkingBeltAirport

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1154 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl         | 61 ++++++++++++++++---
 1 file changed, 52 insertions(+), 9 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index e2152485..0b2b285a 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.12";
+   const string interface_version = "1.13";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -244,6 +244,36 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
 
    };
 
+
+   /**
+    *
+    * Define the baggage status for a position.
+    * Please note: A position is not necessary identical with
+    * a physical belt. Two different positions may in fact be mapped
+    * to just one physical belt.
+    *
+   **/
+   struct baggageStatus
+   {
+
+      /**
+       *
+       * The number of bags here.
+       *
+      **/
+      long              count;
+
+      /**
+       *
+       * Is there a bag, either physically or because of a shared
+       * configuration with a neighbouring position?
+       *
+      **/
+      boolean           occupied;
+
+   };
+
+
    /**
     *
     * Defines the current device status. An object of this kind is
@@ -266,29 +296,42 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       **/
       errorType         errorStatus;
 
-      // The status of the sub devices.
-      moduleStatusList subDeviceStatusList;
+      /**
+       *
+       * The status of the sub devices.
+       *
+      **/
+      moduleStatusList  subDeviceStatusList;
+
+      /**
+       *
+       * Baggage status of the insertion belt.
+       *
+      **/
+      baggageStatus     insertionBelt;
 
       /**
        *
-       * Is there a bag on the insertion belt?
+       * Baggage status of the verification belt.
        *
       **/
-      boolean insertionBeltOccupied;
+      baggageStatus     verificationBelt;
 
       /**
        *
-       * Is there a bag on the verification belt?
+       * Status of the baggage on parking belt, which
+       * can still be controlled by the airline application.
        *
       **/
-      boolean verificationBeltOccupied;
+      baggageStatus     parkingBeltAirline;
 
       /**
        *
-       * Is there a bag on the parking belt?
+       * Status of the baggage on parking belt, which
+       * is now in the responsibility of the airport.
        *
       **/
-      boolean parkingBeltOccupied;
+      baggageStatus     parkingBeltAirport;
 
    };
 
-- 
2.41.0.windows.1

